### PR TITLE
Upgrade mortar, so it gives user an option to either grind or juice.

### PR DESCRIPTION
**Pull request by nikita-chekistov**
## About The Pull Request
So this idea came to me, when I tried to grind Korta nut into Korta Flour in mortar. I failed and my dissapointment was immeasurable. Someone thought this was very smart to JUICE first, and if it's not juiceable, only then grind it.

The solution is presented on the images below. Now you can choose to either grind or juice with the **Stylish** radial menu.
image
Скриншот 02-08-2022 010237
Скриншот 02-08-2022 010233

## Changelog

:cl: SuperDrish
add: Mortar now gives you an option. Do you want to grind or juice your thing?
/:cl:

### CODE REVIEW

**Comment by Microvolnovka19** [comment]
Wait I forgot one thing

**Comment by Fikou** [comment]
why not lmb/rmb?

File: code/modules/reagents/reagent_containers/glass.dm [hunk 1]
```diff
@@ -353,22 +353,30 @@
 			if(user.getStaminaLoss() > 50)
 				to_chat(user, span_warning("You are too tired to work!"))
 				return
+			var/list/choose_options = list(
+				"Grind" = image(icon = 'modular_skyrat/modules/morenarcotics/icons/crack.dmi', icon_state = "cocaine"), //Jesse where is the cocainer
```
Comment (Fikou) [inline code/modules/reagents/reagent_containers/glass.dm:357]: excuse moi

**Comment by Microvolnovka19** [comment]
> why not lmb/rmb?

_I simply dont know how to do so_

**Comment by Time-Green** [comment]
Mortar’s aren’t exactly common usage, putting it on lmb and rmb will probably confuse people when they guess wrong

**Comment by Paxilmaniac** [comment]
the radial is probably better in this case anyways, because its similar functionality to how the reagent grinder machines work

File: code/modules/reagents/reagent_containers/glass.dm [hunk 2]
```diff
@@ -353,22 +353,30 @@
...
+				switch(picked_option) // Should we juice or grind?
+					if("Juice") 
+						if(grinded.juice_results)
+							grinded.on_juice()
+							reagents.add_reagent_list(grinded.juice_results)
+							to_chat(user, span_notice("You juice [grinded] into a fine liquid."))
+							QDEL_NULL(grinded)
+							return
+					if("Grind")
+						if(grinded.reagents)
```
Comment (Kapu1178) [inline code/modules/reagents/reagent_containers/glass.dm:373]:
```suggestion
						if(grinded.grind_results)
```

**Comment by Justice12354** [comment]
In the case of keeping the radial menu, I think that, for consistency's sake, it'd be cool if the icons used were the same as the ones in the reagent grinder.

File: code/modules/reagents/reagent_containers/glass.dm [hunk 3]
```diff
@@ -353,22 +353,30 @@
 			if(user.getStaminaLoss() > 50)
 				to_chat(user, span_warning("You are too tired to work!"))
 				return
+			var/list/choose_options = list(
+				"Grind" = image(icon = 'icons/obj/objects.dmi', icon_state = "ash"),
```
Comment (orthography) [inline code/modules/reagents/reagent_containers/glass.dm:357]:
```suggestion
				"Grind" = image(icon = 'icons/hud/radial.dmi', icon_state = "radial_juice")```

File: code/modules/reagents/reagent_containers/glass.dm [hunk 4]
```diff
@@ -353,22 +353,30 @@
 			if(user.getStaminaLoss() > 50)
 				to_chat(user, span_warning("You are too tired to work!"))
 				return
+			var/list/choose_options = list(
+				"Grind" = image(icon = 'icons/obj/objects.dmi', icon_state = "ash"), 
+				"Juice" = image(icon = 'icons/obj/drinks.dmi', icon_state = "glass_orange")
```
Comment (orthography) [inline code/modules/reagents/reagent_containers/glass.dm:358]:
```suggestion
				"Juice" = image(icon = 'icons/hud/radial.dmi', icon_state = "radial_grind")

**Review by orthography — CHANGES_REQUESTED** [review]
I really like the idea of using sprites for symbolism, but consistency seems important for something like this to avoid confusion.

File: code/modules/reagents/reagent_containers/glass.dm (hunk 4 above)
Comment (Microvolnovka19) [reply code/modules/reagents/reagent_containers/glass.dm:358]: I get the idea, but I'll do it tomorrow.
Also, your in your sugg you've placed the grind.dmi in juice and vice versa

Comment (orthography) [reply code/modules/reagents/reagent_containers/glass.dm:358]: OOPS!

Comment (orthography) [reply code/modules/reagents/reagent_containers/glass.dm:358]: @Microvolnovka19 They should be able to one click now for your convenience.

**Comment by Wallemations** [comment]
> Mortar’s aren’t exactly common usage, putting it on lmb and rmb will probably confuse people when they guess wrong

That's what tooltips are for

**Comment by private-tristan** [comment]
> that's what tooltips are for

I don't like how tooltips look so I have never played with them on because I already know what most items do, I'd rather this be a radial due to, as time-green said, mortars being so rarely used for it to be a lmb/rmb function

**Comment by Microvolnovka19** [comment]
Should be good, but I need to update examples

**Comment by orthography** [comment]
Could you make the note on the description to use alt-click to eject bold? (<b>text</b>)

**Comment by Microvolnovka19** [comment]
Huh. Looks like that's not that easy as I thought it would be

**Comment by orthography** [comment]
> Huh. Looks like that's not that easy as I thought it would be

You accidentally put a page break in the description of the pestle. @Microvolnovka19

**Comment by Wallemations** [comment]
> > that's what tooltips are for
> 
> 
> 
> I don't like how tooltips look so I have never played with them on because I already know what most items do, I'd rather this be a radial due to, as time-green said, mortars being so rarely used for it to be a lmb/rmb function 

Just because you don't like tooltips doesn't mean we shouldn't put them in

**Comment by Microvolnovka19** [comment]
I wish to leave the radial menu because of the reasons:
1. As written upper, its more intuitive
2. I dont know how to do lmb/rmb

And I think it's ready to merge.

**Comment by Microvolnovka19** [comment]
Freaking awesome

**Pull request closed without merging**